### PR TITLE
Add full command-line flags to the reference docs

### LINKED ISSUE #7728: document strict-config and other command line flags
I was recently asked to explain why I was adding `--strict-config` to a pytest ini, "as it appears to be undocumented"

currently command line flags are only documented in the changelog and https://docs.pytest.org/en/stable/_modules/_pytest/main.html

I think we should automatically add the output of `pytest --help` to the docs, eg here:
https://docs.pytest.org/en/stable/usage.html#getting-help-on-version-option-names-environment-variables

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #7728

### CODE REVIEW

File: doc/en/reference.rst [hunk 1]
```diff
@@ -1652,3 +1652,306 @@ passed multiple times. The expected format is ``name=value``. For example::
...
+
+.. _`command-line-flags`:
+
+Command-line Flags
+------------------
+
+All the command-line flags can be obtained by running ``pytest --help``::
+
+    $ pytest --help
+    usage: pytest [options] [file_or_dir] [file_or_dir] [...]
```
Comment (nicoddemus) [inline doc/en/reference.rst:1665]: This line an below were generated by `regendoc`.

File: doc/en/reference.rst [hunk 2]
```diff
@@ -1652,3 +1652,306 @@ passed multiple times. The expected format is ``name=value``. For example::
...
+                            log file logging level.
+      --log-file-format=LOG_FILE_FORMAT
+                            log format as used by the logging module.
+      --log-file-date-format=LOG_FILE_DATE_FORMAT
+                            log date format as used by the logging module.
+      --log-auto-indent=LOG_AUTO_INDENT
+                            Auto-indent multiline messages passed to the logging
+                            module. Accepts true|on, false|off or an integer.
+
+    Hypothesis:
```
Comment (asottile) [inline doc/en/reference.rst:1858]: this includes our plugins it looks like so we probably need to filter those out somehow?

Comment (nicoddemus) [reply doc/en/reference.rst:1858]: Ahh my bad. I ran regendoc manually. I will update this.

**Review by asottile — APPROVED** [review]

**Pull request merged**